osgp/platform/osgp-adapter-domain-smartmetering/src/test/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/EventMessageDataContainerMappingTest.java

```diff
@@ -56,6 +57,8 @@ public void testWithEmptyList() {
     public void testWithFilledList() {
         // build test data
         final EventDto event = new EventDto(new DateTime(), new Integer(1), new Integer(2), "STANDARD_EVENT_LOG");
+        event.setEventTypeDto(EventTypeDto.POWER_FAILURE);
```

Comment:
eventType could also be added to the constructor, to prevent mixture of setters and constructor instantiation.

Reply:
We could do this, but than we get constructor overloading, and we need a setter, because we add the EventType afterwards in the EventService, based on the eventCode